comms/RepositoryImpl.h

```diff
@@ -283,7 +283,8 @@ class CRepositoryBase : public IRepository, public CUnknown
         GetAttributes(module.c_str(), attrs, true);
         for(IAttributeVector::const_iterator itr = attrs.begin(); itr != attrs.end(); ++itr)
         {
-            results.push_back(itr->get()->GetLabel());
+            if (boost::algorithm::iequals(ExtensionWithoutDot(fileExt), ATTRIBUTE_TYPE_ECL) && itr->get()->GetType() == CreateIAttributeECLType())
```

Comment:
Potential memory leak with:  ```CreateIAttributeECLType``` - you can use the type directly now without this call.
What is the ```boost::algorithm::iequals(ExtensionWithoutDot(fileExt), ATTRIBUTE_TYPE_ECL)``` test doing?  Should it not be:  ```boost::algorithm::iequals(ExtensionWithoutDot(itr->get()->GetLabel()), ATTRIBUTE_TYPE_ECL)``` or similar?

Reply:
@GordonSmith I believe this logic to be correct and the menu does not display the extensions. It only displays ecl files and subfolders. The variable `fileExt` is the file extension from the attribute file where the user has just typed the ".". This test is to make sure we only pop up the autocomplete when the user is inside an ECL file. As for the other test, there should be no memory leak given this is cached value and will reuse it each time it is called. The other way to check would be to do a string comparison on the type. Not sure which is more efficient however, `== CreateIAttributeECLType()` is used in the same way in EclDlgAttribute.cpp on line 130 and in one other place.

Reply:
Shouldn't the file type check be done before making the call and not in the middle of an iteration?

Also I have been converting those other CreateIAttibuteECLType calls to straight comparisons as I come across them:  ```itr->get()->IsTypeOf(ATTRIBUTE_TYPE_ECL))```

Reply:
Yes, you are right. Should be done before the loop. Gotcha on the comparison conversion.